src/test/user.spec.js

```
@@ -0,0 +1,204 @@
+      const newUser = {
+        email: 'timi.tejumola@andelahub.com',
+        username: 'timiosh',
+        password: 'password'
+      };
+      chai
+        .request(app)
+        .post('/api/v1/auth/signup')
+        .send(newUser)
+        .end((err, res) => {
+          expect(res.status).to.equal(400);
+          expect(res.body).to.be.an('object');
+          expect(res.body).to.have.property('status');
+          expect(res.body.message).to.equal(
+            'You already have an account. Kindly check your email address for a verification link.'
+          );
+          done();
+        });
+    });
+
+    it('it should throw an error when user with an active account tries to signup', done => {
+      const newUser = {
+        email: 'jonsnow@got.com',
+        username: 'timiosh',
+        password: 'password'
+      };
+      chai
+        .request(app)
+        .post('/api/v1/auth/signup')
+        .send(newUser)
+        .end((err, res) => {
+          expect(res.status).to.equal(400);
+          expect(res.body).to.be.an('object');
+          expect(res.body).to.have.property('status');
+          expect(res.body.message).to.equal(
+            'User with this email address already exist'
+          );
+          done();
+        });
+    });
+  });
+
+  describe('POST /auth/login', () => {
+    it('it should throw an error if email address is invalid', done => {
+      const user = {
+        email: 'sholabolagmail.com',
+        password: 'password'
+      };
+      chai
+        .request(app)
+        .post('/api/v1/auth/login')
+        .send(user)
+        .end((err, res) => {
+          expect(res.status).to.equal(400);
+          expect(res.body).to.be.an('object');
+          expect(res.body).to.have.property('status');
+          expect(res.body.message).to.equal('email must be a valid email');
+          done();
+        });
+    });
+
+    it('it should throw an error if password is not up to 8', done => {
```

`it should throw an error if password is less than 8 characters long` - Be explicit.

Do we have a maximum password length as well?

---

Currently, the password column can take up to 255 characters. I don't think it's necessary to have a maximum password length since it will be hashed before it is inserted into the database.